sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java

```diff
@@ -611,8 +622,12 @@ private boolean ensureValidKeyAvailable() {
         boolean keyAvailableLocally = true;
         if (this.key == null) {
```

Comment:
do we want a check for `keyCollection` ?

Reply:
added the check